src/azure-cli/azure/cli/command_modules/acs/azurecontainerstorage/_validators.py

```diff
@@ -0,0 +1,275 @@
+            )
+
+        if (is_ephemeralDisk_localssd_enabled or is_ephemeralDisk_nvme_enabled) and \
+           storage_pool_type == CONST_STORAGE_POOL_TYPE_EPHEMERAL_DISK:
+            ephemeral_disk_type_installed = CONST_STORAGE_POOL_OPTION_SSD if \
+                                            is_ephemeralDisk_localssd_enabled else \
+                                            CONST_STORAGE_POOL_OPTION_NVME 
+            raise ArgumentUsageError(
+                'Invalid --enable-azure-container-storage value. '
+                'Azure Container Storage is already enabled for storagepool type '
+                f'{CONST_STORAGE_POOL_TYPE_EPHEMERAL_DISK} and option {ephemeral_disk_type_installed} '
+                'in the cluster.'
+            )
+    else:
+        _validate_nodepool_names(nodepool_list, agentpool_details)
+
+
+# _Validate_storage_pool_size validates that the storage_pool_size is
+# string of a combination of a float number immediately followed by
+# Ti or Gi e.g. 2Ti, 512Gi, 1.5Ti. The function also validates that the
+# minimum storage pool size for an elastic san storagepool should be 1Ti.
+def _validate_storage_pool_size(storage_pool_size, storage_pool_type):
+    if storage_pool_size is not None:
+        pattern = r'^\d+(\.\d+)?[GT]i$'
+        match = re.match(pattern, storage_pool_size)
+        if match is None:
+            raise ArgumentUsageError(
+                'Value for --storage-pool-size should be defined '
+                'with size followed by Gi or Ti e.g. 512Gi or 2Ti.'
+            )
+        if storage_pool_type == CONST_STORAGE_POOL_TYPE_ELASTIC_SAN:
+            pool_size_qty = float(storage_pool_size[:-2])
+            pool_size_unit = storage_pool_size[-2:]
+
+            if (
+                (pool_size_unit == "Gi" and pool_size_qty < 1024) or
+                (pool_size_unit == "Ti" and pool_size_qty < 1)
+            ):
+                raise ArgumentUsageError(
+                    'Value for --storage-pool-size must be at least 1Ti when '
+                    '--enable-azure-container-storage is elasticSan.'
+                )
+
+        elif storage_pool_type == CONST_STORAGE_POOL_TYPE_EPHEMERAL_DISK:
+            logger.warning(
+                'Storage pools using Ephemeral disk use all capacity available on the local device. '
+                ' --storage-pool-size will be ignored.'
+            )
+
+
+# _validate_nodepool_names validates that the nodepool_list is a comma separated
+# string consisting of valid nodepool names i.e. a lower alphanumeric
+# characters and the first character should be lowercase letter.
+def _validate_nodepool_names(nodepool_names, agentpool_details):
+    pattern = r'^[a-z][a-z0-9]*(?:,[a-z][a-z0-9]*)*$'
+    if re.fullmatch(pattern, nodepool_names) is None:
+        raise InvalidArgumentValueError(
+            "Invalid --azure-container-storage-nodepools value. "
+            "Accepted value is a comma separated string of valid nodepool "
```

Comment:
consistency: "node pools" 
should be two distinct words in English, consistent with the AKS docs:
https://learn.microsoft.com/en-us/azure/aks/create-node-pools

in the command it is correct to use the form `nodepool`